src/System.Management.Automation/engine/regex.cs

```diff
@@ -148,7 +146,26 @@ private void Init()
             {
                 if (Pattern.Length == 1 && Pattern[0] == '*')
                 {
-                    _isMatch = _ => true;
+                    _isMatch = s_matchAll;
+                }
+                else if (Pattern.IndexOfAny(s_specialChars) == -1)
+                {
+                    // No special characters present in the pattern, so we can just do a string comparison.
+                    StringComparison stringComparison;
+                    if (Options.HasFlag(WildcardOptions.IgnoreCase))
```

Comment:
Given the depth of this `if` statement and to @iSazonov's point about readability, I think this method is now better defined as:

```csharp
private void Init()
{
    if (_isMatch != null) { return; }

    if (Pattern.Length == 1 && Pattern[0] == '*')
    {
        _isMatch = s_matchAll;
        return;
    }

    if (Pattern.IndexOfAny(s_specialChars) == -1)
    {
        // String comparison definition
        _isMatch = str => string.Equals(str, Pattern, stringComparison);
       return;
    }

    _isMatch = new WildcardPatternMatcher(this).IsMatch;
}
```

Really though, `Init()`should just be a `Lazy<Predicate<string>>` private property

Reply:
This does look better. Will make the change.